src/main/java/com/collicode/propertytracker/api/RoomApiController.java

```diff
@@ -24,13 +27,38 @@ public ResponseEntity<?> createRoom(@RequestHeader("X-RequestId") String request
       return ResponseEntity.status(500).body(ex.getMessage());
     }
   }
-  @GetMapping("/{apartmentId}")
+  @GetMapping("/{roomId}")
   public ResponseEntity<?> fetchRoomsByApartmentId(@RequestHeader("X-RequestId") String requestId,@PathVariable long apartmentId) {
     try {
       return ResponseEntity.ok().body(roomService.fetchRoomsByApartmentId(apartmentId));
     } catch (Exception ex) {
       return ResponseEntity.status(500).body(ex.getMessage());
     }
   }
+  @PutMapping("/{roomId}")
+  public ResponseEntity<?> updateRoom(@RequestHeader("X-RequestId") String requestId,
+                                           @PathVariable long roomId,
+                                           @RequestBody RoomUpdateRequestDTO roomUpdateRequestDTO) {
+    try {
+      roomService.updateRoom(roomId, roomUpdateRequestDTO);
+      return ResponseEntity.ok().body(roomUpdateRequestDTO);
+    }catch (EntityNotFoundException ex){
+      return ResponseEntity.status(404).body(ex.getMessage());
+    }catch (StorageException ex){
+      return ResponseEntity.status(500).body(ex.getMessage());
+    }
+  }
+  @DeleteMapping("/{roomId}")
+  public ResponseEntity<?> deleteRoom(@RequestHeader("X-RequestId") String requestId,
```

Comment:
also deleting room add apartmentId to indicate you are deleting a room for a particular apartment